lib/availability_resolver.js

```diff
@@ -99,6 +99,16 @@ class AvailabilityResolver {
 
         // If it's in ReCAP
         if (isInRecap) {
+          // If this serialization is a result of a hold request initializing, we want
+          // to double check the recap customer code in SCSB
+          if (itemUri){
+            const mostRecentCustomerCode = await this.SCSBRestClient.search({ fieldValue: barcode, fieldName: 'Barcode' })
```

Comment:
I think you may need to pull the actual customer code out of some nested property of the response obtained here. We should prob add a `lib/scsb-client.js` file, which exports a `customerCodeByBarcode` method, [which essentially runs this](https://github.com/NYPL/discovery-api/blob/d955b3ff85c4363900c6bfc85c192609b740c8f4/lib/delivery-locations-resolver.js#L106-L132), encapsulating all of the error handling and messy logic about extracting customer code from the two places it might live. And then we can just call that `scsbClient.customerCodeByBarcode` method here. We should then have the `DeliveryLocationsResolver` also use that util method (although that need not happen with this PR necessarily)